applications/PfemFluidDynamicsApplication/custom_processes/fix_free_velocity_on_nodes_process.cpp

```diff
@@ -0,0 +1,63 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics PfemFluidDynamics Application
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo Velazquez
+//                   Carlos Eulogio Flores
+//
+
+#include "custom_processes/fix_free_velocity_on_nodes_process.h"
+#include "includes/define.h"
+#include "includes/kratos_flags.h"
+
+namespace Kratos {
+
+FixFreeVelocityOnNodesProcess::FixFreeVelocityOnNodesProcess(
+    ModelPart& rModelPart,
+    const int rFreeOrFix)
+    : mrModelPart(rModelPart),
+      mFreeOrFix(rFreeOrFix)
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FixFreeVelocityOnNodesProcess::Execute() 
+{
+    const auto& it_node_begin = mrModelPart.NodesBegin();
+    #pragma omp parallel for
+    for (int i = 0; i < static_cast<int>(mrModelPart.Nodes().size()); i++) {
+        auto it_node = it_node_begin + i;
+        if (mFreeOrFix == 0) {
+            it_node->Fix(VELOCITY_X);
+            it_node->Fix(VELOCITY_Y);
+            it_node->Fix(VELOCITY_Z);
+
+        } else {
+            it_node->Free(VELOCITY_X);
+            it_node->Free(VELOCITY_Y);
+            it_node->Free(VELOCITY_Z);
+
+        }
+    }
+    if (mFreeOrFix == 0) 
+    {
+        KRATOS_INFO("FixFreeVelocityOnNodesProcess") << " FIXED VELOCITY VALUES FOR THE NODES " << std::endl;
+    }
+    else
+    {
+        KRATOS_INFO("FixFreeVelocityOnNodesProcess") << " FREED VELOCITY VALUES FOR THE NODES " << std::endl;
+    }
+}
```

Comment:
I recommend using https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/variable_utils.h#L992-L1025

It does loop the nodes 3 times, but I think it is better to use the default implementation

Alternatively if you want to keep the current implementation then please use the `ParallelUtilities` => search for `block_for_each` in the code, you will get lots of examples

Reply:
I saw the examples and changed it accordingly.